Sources/Shared/ImageView+Imaginary.swift

```diff
@@ -45,42 +45,54 @@ extension ImageView {
   }
 
   fileprivate func fetchFromNetwork(url: URL, configuration: Configuration = Configuration.default, completion: Completion? = nil) {
-    fetcher = Fetcher(url: url)
-    fetcher?.start(configuration.preprocess) { [weak self] result in
+    imageDownloader = ImageDownloader(url: url)
+    imageDownloader?.start(configuration.preprocess) { [weak self] result in
       guard let `self` = self else {
         return
       }
 
       switch result {
-      case let .success(image, bytes):
-        configuration.track?(url, nil, bytes)
+      case .image(let image):
+        configuration.track?(url, nil)
         configuration.transitionClosure(self, image)
         let wrapper = ImageWrapper(image: image)
         configuration.imageStorage?.async.setObject(wrapper, forKey: url.absoluteString, expiry: nil) { _ in
           // Don't care about result for now
         }
-        completion?(image)
-      case let .failure(error):
-        configuration.track?(url, error, 0)
+        completion?(.image(image))
+      case .error(let error):
+        completion?(.error(error))
+        configuration.track?(url, error)
       }
 
       configuration.postConfigure?(self)
     }
   }
 
-  var fetcher: Fetcher? {
+  var imageDownloader: ImageDownloader? {
     get {
       let wrapper = objc_getAssociatedObject(self, &Capsule.ObjectKey) as? Capsule
-      let fetcher = wrapper?.concept as? Fetcher
-      return fetcher
+      let imageDownloader = wrapper?.concept as? ImageDownloader
+      return imageDownloader
     }
-    set (fetcher) {
+    set (imageDownloader) {
       var wrapper: Capsule?
-      if let fetcher = fetcher {
-        wrapper = Capsule(concept: fetcher)
+      if let imageDownloader = imageDownloader {
+        wrapper = Capsule(concept: imageDownloader)
       }
       objc_setAssociatedObject(self, &Capsule.ObjectKey,
                                wrapper, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC)
     }
   }
 }
+
+
+/// Used to associate ImageDownloader with ImageView
+fileprivate class Capsule: NSObject {
+  static var ObjectKey = 0
+  let concept: Any
+
+  init(concept: Any) {
```

Comment:
concept? 🤔

Reply:
@vadymmarkov It's from old code. But this `Capsule` will be removed soon